frontend/src/components/ShootWorkers/ManageWorkers.vue

```diff
@@ -110,15 +116,56 @@ export default {
     allZones () {
       return this.zonesByCloudProfileNameAndRegion({ cloudProfileName: this.cloudProfileName, region: this.region })
     },
+    currentZonesWithNetworkConfigInShoot () {
+      return map(this.currentZonesNetworkConfiguration, 'name')
+    },
+    currentFreeNetworks () {
+      if (this.isNewCluster) {
+        return this.allZones.length
+      }
+      if (!cloudProviderSupportsAddingZonesAfterCration(this.cloudProviderKind)) {
+        return 0
+      }
+      return findFreeNetworks(this.currentZonesNetworkConfiguration, this.existingWorkerCIDR, this.cloudProviderKind, this.allZones.length)
+    },
     availableZones () {
+      if (this.isNewCluster) {
+        return this.allZones
+      }
       // Ensure that only zones can be selected, that have a network config in providerConfig (if required)
-      // Can be removed when gardener supports to change network config afterwards
-      // --> Allow adding zones post-shoot creation PR: https://github.com/gardener/gardener/pull/1587
-      const zonesWithNetworkConfigInShoot = map(this.zonesNetworkConfiguration, 'name')
-      if (!isEmpty(zonesWithNetworkConfigInShoot)) {
-        return zonesWithNetworkConfigInShoot
+      // or that free networks are available to use more zones
+      if (!isEmpty(this.currentZonesWithNetworkConfigInShoot)) {
+        if (!this.currentFreeNetworks.length) {
+          return sortBy(this.currentZonesWithNetworkConfigInShoot)
+        }
+      }
```

Comment:
don't understand

Reply:
refactored code, maybe now it is easier to understand